google/ggrc-core
# GGRC-7284 Replace Deferred with Promise

# Issue description

In scope of the migration to CanJS 3 also jQuery was updated to version 3. In version 3 some async changes were introduced for Deferred. It allows to start replacement for Deferred to Promise usages.

# Steps to test the changes

1. GGRC gdrive picker launcher & attach button & uploadFiles() method.
1.1 Launch GGRC.
1.2 Create new audit and assessment.
1.3 Assign folder for audit.
1.4 Attach some files to folder and upload.
1.5 Expected: It works.
2. Confirm edit button.
2.1 Launch GGRC.
2.2 Open assessment.
2.3 Make it completed.
2.4 Click "Edit Answers" button and confirm moving.
2.5 Expected: It works.
3. Create document button.
3.1 Launch GGRC.
3.2 Create or open existing data assets object.
3.3 Click "Attach button".
3.4 Click "attach new" on opened modal.
3.5 Select some files and upload.
3.6 Expect: It works.
4. Workflow utils.
4.1 Launch GGRC.
4.2 Create new workflow.
4.3 Setup task group and activate cycle.
4.4 Expected: It works.
#### batchRequests util. 
1. widgets-utils:
1.1 Open any object.
1.2 Expected: It works.
2. tree-view-utils:
2.1 Open All objects tab and chose any type.
2.2 Click expand tree from object settings popup.
2.3 Expected: It works correctly.
3. related-objects:
3.1 Open program with proposals.
3.2 Open Change proposals tab.
3.3 Expected: It works correctly.
3.4 Open assessment with mapped issues.
3.5 Open related issues tab.
3.6 Expected: It works correctly.
4. revision-log:
4.1 Open any object and open change log tab.
4.2 Expected: It works correctly.
5. issue-unmap-item:
5.1 Open a issues which was mapped to assessment.
5.2 Open the assessment and chose unmap in popup. 
5.3 Expected: List of mapped assessments and audits loads correctly.
6. assessment-info-pane:
6.1 Open audit with mapped controls.
6.2 Create new assessment for this audit.
6.3 Open created assessment.
6.4 Map control to this assessment.
6.5 Expect: It works.
7. related-comments:
7.1 Open a assessment with related assessment. 
7.2 Click more info. 
7.3 Open a comments tab. 
7.2 Expected: It works correctly.
8. mapped-counter:
8.1 Open a fold-out menu with objects. 
8.2 Hover on any object. 
8.3 Expected: Counts of mapped objects loads correctly.




# Solution description

Replace Deferred with Promise.

# Sanity checklist




# PR Review checklist
